exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -269,11 +291,25 @@ func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.Ti
 		return errs
 	}
 
-	input := make(chan *prompb.WriteRequest, len(requests))
-	for _, request := range requests {
-		input <- request
+	if !prwe.walEnabled() {
+		return prwe.actualExport(ctx, requests)
 	}
-	close(input)
+
+	// Otherwise, serialize the requests to the WAL.
+	if err := prwe.writeToWAL(requests); err != nil {
+		errs = append(errs, consumererror.Permanent(err))
+	}
+	return errs
+}
+
+func (prwe *PrwExporter) actualExport(ctx context.Context, requests []*prompb.WriteRequest) (errs []error) {
```

Comment:
Can we name this something like exportWriteRequests?

Reply:
Done, thanks!